### ISSUE TITLE
Why do workspaces have to be restarted to apply the autostop setting?

### ISSUE THREAD

**Issue by ammario** [coder/coder#14242]
This is an unfortunate UX:

![image](https://github.com/user-attachments/assets/1387ec9d-8192-4196-8c77-c23e9a0bfb46)

Instead, we could set the auto-stop deadline to now+ttl?

**Comment by stirby** [comment]
We point users to the [`coder override-stop`](https://coder.com/docs/cli/schedule_override-stop) command when this is raised. I believe there are some limitations in applying scheduling changes while the workspace is running. 

If this can be corrected by applying the `now+ttl` as you suggested, we should make the change. It's commonly requested by customers who have long-running ML jobs and need to suspend the autostop on the fly.

@f0ssel or @johnstcn, given your recent contribution to activity reporting and scheduling features, could you weigh in on the feasibility of this?

**Comment by johnstcn** [comment]
It would be more 'correct' to update deadline to workspace_start + TTL. However, if the new TTL is less than the old TTL, this could potentially result in running workspaces being stopped on the next autostop 'tick'. Setting to now plus TTL does make this slightly easier although we would still want to keep this bounded by max_deadline.

This is also a step towards replacing `workspaces.deadline` with `workspace_builds.completed_at+workspaces.ttl`. We had originally planned to do this as part of the [workspace activity refactor](https://github.com/coder/coder/issues/13279) but we found it would cause scheduling behavioural changes we weren't ready to commit to at the time.

**Comment by matifali** [comment]
@johnstcn is this still the case that we need to restart workspace to apply scheduling .

**Comment by johnstcn** [comment]
> [@johnstcn](https://github.com/johnstcn) is this still the case that we need to restart workspace to apply scheduling .

Yes, and my previous comment still stands - are we OK with changing this behaviour such that users (or admins) can essentially cause workspaces to stop underneath users' feet?

**Comment by matifali** [comment]
I think no. May be in the next quite hours window if set.

Ideally we would need to find a way to to update this without restarting the workspace.
